src/chain.rs

```diff
@@ -224,41 +138,199 @@ impl proptest::arbitrary::Arbitrary for Chain {
     fn arbitrary_with((): ()) -> Self::Strategy {
         use proptest::prelude::*;
         prop_oneof![
-            any::<Selector>().prop_map(move |sel| Chain::Named(sel.select(NamedChain::iter()))),
-            any::<u64>().prop_map(Chain::Id),
+            any::<Selector>().prop_map(move |sel| Self::from_named(sel.select(NamedChain::iter()))),
+            any::<u64>().prop_map(Self::from_id),
         ]
     }
 }
 
+impl Chain {
```

Comment:
can we move this below the struct def?

Reply:
I prefer trait impls blocks before main impl, that's how I did it in all of core

Reply:
I'm used to above trait impls -.-

but I don't mind